src/twisted/web/client.py

```diff
@@ -1770,7 +1770,10 @@ def connectionLost(self, reason):
                 )
             )
         else:
-            self.deferred.errback(reason)
+            try:
+                self.deferred.errback(reason)
+            except defer.AlreadyCalledError:
+                pass
```

Comment:
Note that we should also have a similar code for the case of `PartialDownloadError`  ... line 1767

I don't know how else we can do here.

It would be nice to ignore the error only if the deferred was cancelled ... but I don't know if there is an easy way.

What I am trying to have here, is some way to make sure that you don't accidentally try to raise multiple errors for the same deferred 

```suggestion
            except defer.AlreadyCalledError:
                # Somehow the error was already triggered.
                # Maybe the deferred was cancelled by the user.
                pass
```

Reply:
1. I'm not sure that we require the same code for PartialDownloadError (line 1767) because the code should not be called after cancel.
Yes. In current Twisted code it's possible, because cancel does't abort connection. It mentioned here: https://github.com/twisted/twisted/issues/8227#issuecomment-1168288396
I'm going to fix it in another PR.
2. I think that ignored error only for canceled case is implemented in my commit https://github.com/twisted/twisted/pull/12502/commits/a0767b72893d70e7fa54667698190eaac3d3aa59